File: snakebot-c#/src/Cygni.Snake.Client/PlayerRegistered.cs
namespace Cygni.Snake.Client
{
    public class PlayerRegistered : GameEvent
    {
        public string Name { get; }

        public GameSettings GameSettings { get; }

        public string GameMode { get; }

        public override string Type => MessageType.PlayerRegistered;

        public PlayerRegistered(string name, GameSettings gameSettings, string gameMode, string gameId, string receivingPlayerId) 
            : base(gameId, receivingPlayerId)
        {
            Name = name;
            GameSettings = gameSettings;
            GameMode = gameMode;
        }
    }
}

File: domain/mamba/playerRegistered.js
const GameSettings = require('./gameSettings.js');

function PlayerRegistered(gameId, gameMode, receivingPlayerId, name, settings) {
    const type = 'se.cygni.snake.api.response.PlayerRegisteredEvent';
    var gameId = gameId;
    var gameMode = gameMode;
    var receivingPlayerId = receivingPlayerId;
    const playerName = name;
    var settings = settings;

    const toString = function () {
        return `<Type:${type}gameId:${gameId}, gameMode:${gameMode}, receivingPlayerId:${receivingPlayerId
        }playerName:${playerName}, settings:${settings}>`;
    };

    function getReceivingPlayerId() {
        return receivingPlayerId;
    }

    function getGameId() {
        return gameId;
    }

    function updateGameId(theGameId) {
        gameId = theGameId;
    }

    function getGameMode() {
        return gameMode;
    }

    function getPlayerName() {
        return playerName;
    }

    function getSettings() {
        return settings;
    }

    return Object.freeze({
        type,
        toString,
        getPlayerId: getReceivingPlayerId,
        getGameId,
        getGameMode,
        getPlayerName,
        getGameSettings: getSettings,
        updateGameId
    });
}

function create(data) {
    return PlayerRegistered(
        data.gameId,
        data.gameMode,
        data.receivingPlayerId,
        data.name,
        GameSettings.create(data.gameSettings)
    );
}

exports.new = PlayerRegistered;
exports.create = create;
exports.type = PlayerRegistered().type;
